Objects/unicodeobject.c

```diff
@@ -6802,17 +6802,21 @@ _PyUnicode_DecodeUnicodeEscapeStateful(const char *s,
         unsigned char c = *first_invalid_escape;
         if ('4' <= c && c <= '7') {
             if (PyErr_WarnFormat(PyExc_DeprecationWarning, 1,
-                                 "invalid octal escape sequence '\\%.3s'",
-                                 first_invalid_escape) < 0)
+                                 "\"\\%.3s\" is an invalid octal escape sequence. "
```

Comment:
Currently, an escape sequence in a (Unicode) string like `'\407'` is interpreted as `'ć'`, i.e. the value is converted from octal and looked up as a Unicode code point anyway. It's not clear to me why this has to be considered invalid syntax at all (unless octal escapes were removed completely) - considering that the reporting for a `\U` escape with value >= 0x110000 is totally different (still a `SyntaxError`, but presented as if it were a codec issue). Those values 0x400 - 0x7ff are perfectly within range for Unicode, which perfectly well defines their meanings, and strings are *not* byte sequences; so why restrict it?

Reply:
The purpose of this PR is simply to modify the existing warning message. If you intend to change Python behavior perhaps it'd be worth creating a new PR?